### PR TITLE
KAFKA-18100: `Using` block suppresses all errors

### PR DESCRIPTION
**Pull request by TaiJuWu**
JIRA: https://issues.apache.org/jira/browse/KAFKA-18100/

Replace `Using` with `Using.resource` and fix fail test

Co-authored-by: frankvicky [kitingiao@gmail.com](mailto:kitingiao@gmail.com)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@TaiJuWu Could you replace all `Using` by "try-release" ?

**Comment by chia7712** [comment]
@frankvicky Could you please a PR to fix the ``CoordinatorLoaderImplTest` first?

**Comment by frankvicky** [comment]
Hi @chia7712 
I have file a PR for it #17954

**Comment by TaiJuWu** [comment]
fail test:
```
FAILED ❌ CoordinatorLoaderImplTest > testDeserializationErrorFailsTheLoading()
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testTransactionalProducerWithAuthenticationFailure(String, String).quorum=kraft.groupProtocol=classic
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testManualAssignmentConsumerWithAuthenticationFailure(String, String).quorum=kraft.groupProtocol=classic
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testProducerWithAuthenticationFailure(boolean).quorum=kraft.isIdempotenceEnabled=true
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testProducerWithAuthenticationFailure(boolean).quorum=kraft.isIdempotenceEnabled=false
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testManualAssignmentConsumerWithAutoCommitDisabledWithAuthenticationFailure(String, String).quorum=kraft.groupProtocol=classic
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testKafkaAdminClientWithAuthenticationFailure(String, String).quorum=kraft.groupProtocol=classic
FAILED ❌ SaslClientsWithInvalidCredentialsTest > testConsumerWithAuthenticationFailure(String, String).quorum=kraft.groupProtocol=classic```

**Comment by chia7712** [comment]
@TaiJuWu `RaftManagerTest` and `KafkaServerTestHarness` are using `Using`. Could you please fix them as well?

**Comment by TaiJuWu** [comment]
> RaftManagerTest



> RaftManagerTest

Both of them use `Using.resource` and the method can throw Exception so there is no issue.

**Comment by chia7712** [comment]
@TaiJuWu Could you please use `Using.resource` to minimize the changes?

File: core/src/test/scala/unit/kafka/coordinator/group/CoordinatorLoaderImplTest.scala [hunk 1]
```diff
@@ -312,7 +312,8 @@ class CoordinatorLoaderImplTest {
         .thenThrow(new RuntimeException("Error!"))
 
       val ex = assertFutureThrows(loader.load(tp, coordinator), classOf[RuntimeException])
-      assertEquals("Error!", ex.getMessage)
+
```
Comment (TaiJuWu) [inline core/src/test/scala/unit/kafka/coordinator/group/CoordinatorLoaderImplTest.scala:315]: fix CoordinatorLoaderImplTest > testDeserializationErrorFailsTheLoading()
author: @frankvicky

File: core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala [hunk 2]
```diff
@@ -74,6 +75,8 @@ class SaslClientsWithInvalidCredentialsTest extends AbstractSaslTest {
   override def setUp(testInfo: TestInfo): Unit = {
     startSasl(jaasSections(kafkaServerSaslMechanisms, Some(kafkaClientSaslMechanism), Both,
       JaasTestUtils.KAFKA_SERVER_CONTEXT_NAME))
+    val superuserLoginContext = jaasAdminLoginModule(kafkaClientSaslMechanism)
```
Comment (TaiJuWu) [inline core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala:78]: Fix `SaslClientsWithInvalidCredentialsTest`

**Comment by chia7712** [comment]
@TaiJuWu Could you please fix https://github.com/apache/kafka/blob/434fe7c26e31a6cfeb5e17e83fb52a260eac398e/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala#L2391?

**Comment by TaiJuWu** [comment]
> @TaiJuWu Could you please fix
> 
> https://github.com/apache/kafka/blob/434fe7c26e31a6cfeb5e17e83fb52a260eac398e/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala#L2391
> 
> ?

Done. Thanks for double check!

**Review by chia7712 — APPROVED** [review]
LGTM

Loop the failed test 50 times, all pass. open https://issues.apache.org/jira/browse/KAFKA-18105 to trace it

**Comment by TaiJuWu** [comment]
Since this issue exist in 3.9 and it related our test infra, I back port to 3.9 branch https://github.com/apache/kafka/pull/17970

**Pull request merged**